File: 404. Sum of Left Leaves/solution.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
void dfs(TreeNode* current, int &amount){
  if (current->left != NULL){
    if (current->left->left == NULL && current->left->right == NULL){
      amount += current->left->val;
    }
    else{
      dfs(current->left, amount);
    }
  }
  if (current->right != NULL){
    dfs(current->right, amount);
  }
}
public:
  int sumOfLeftLeaves(TreeNode* root) {
    int total = 0;
    dfs(root, total);
    return total;
  }
};

File: 404. Sum of Left Leaves/solution.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var sumOfLeftLeaves = function(root) {
  let total = 0;

  function dfs(current, amount){
    if (current.left != undefined){
      if (current.left.left == undefined && current.left.right == undefined){
        amount += current.left.val;
      }
      else{
        amount = dfs(current.left, amount);
      }
    }
    if (current.right != undefined){
      amount = dfs(current.right, amount);
    }
    return amount;
  }

  return dfs(root, total);
};